File: cloud/gateway-server/src/main/java/pers/acp/admin/gateway/repo/RouteLogRepository.java
package pers.acp.admin.gateway.repo;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
import pers.acp.admin.gateway.entity.RouteLog;

/**
 * @author zhang by 21/05/2019
 * @since JDK 11
 */
public interface RouteLogRepository extends JpaSpecificationExecutor<RouteLog>, JpaRepository<RouteLog, String> {
}

File: cloud/log-server/src/main/kotlin/pers/acp/admin/log/repo/RouteLogRepository.kt
package pers.acp.admin.log.repo

import pers.acp.admin.common.base.BaseRepository
import pers.acp.admin.log.entity.RouteLog
import java.util.*

/**
 * @author zhang by 21/05/2019
 * @since JDK 11
 */
interface RouteLogRepository : BaseRepository<RouteLog, String> {

    fun findByLogIdAndRequestTime(logId: String, requestParam: Long): Optional<RouteLog>
}
